File: backend-java/src/main/java/com/backend/java/backendjava/adapters/CardAdapter.java
package com.backend.java.backendjava.adapters;

import com.backend.java.backendjava.dtos.requests.CardRequest;
import com.backend.java.backendjava.dtos.responses.CardResponse;
import com.backend.java.backendjava.entities.Card;
import org.springframework.data.domain.Page;

import java.util.List;

public interface CardAdapter {
    CardResponse createCard(CardRequest request);
    Page<CardResponse> readCard();
    CardResponse updateCardById(Long id, CardRequest request);
    void deleteCardById(Long id);
}

File: backend-kotlin/src/main/kotlin/com/backend/kotlin/backendkotlin/adapters/CardAdapter.kt
package com.backend.kotlin.backendkotlin.adapters

import com.backend.kotlin.backendkotlin.dtos.requests.CardRequest
import com.backend.kotlin.backendkotlin.dtos.responses.CardResponse
import org.springframework.data.domain.Page


interface CardAdapter {
    fun createCard(request: CardRequest?): CardResponse?
    fun readCard(): Page<CardResponse?>?
    fun updateCardById(id: Long?, request: CardRequest?): CardResponse?
    fun deleteCardById(id: Long?)
}